### DISCUSSION TITLE
When orderBy raw or other raw queries?

### DISCUSSION THREAD

**Discussion by fobaz** [prisma/prisma#18280] [Q&A] [answered]
## Problem
There is no way to order by raw query except to write the whole query as raw. Not being able to write raw sql for some parts of a query is disgustingly limiting and kinda discourages using Prisma in some bigger applications.

## Suggested solution
Add orderByRaw or something similar. Maybe, for a quick and dirty hack, it's possible to extend Prisma to add custom raw sql to Prisma's generated query? Or to somehow be able to modify Prisma's generated queries?

## Additional context
There are multiple other issues related to this. 
https://github.com/prisma/prisma/issues/6574 https://github.com/prisma/prisma/issues/5560 https://github.com/prisma/prisma/issues/17023 https://github.com/prisma/prisma/issues/12279 https://github.com/prisma/prisma/issues/8068 https://github.com/prisma/prisma/issues/6689 https://github.com/prisma/prisma/issues/5848 https://github.com/prisma/prisma/issues/5761

**Comment by nurul3101** [comment, marked as answer]
Hello @fobaz 👋 

Thank you for comprising a list of GitHub Issues that are related to supporting partial raw queries.
I'll share this internally with our team so we can consider prioritising this.
We will update our [roadmap](https://prismaio.notion.site/Prisma-ORM-Roadmap-50766227b779464ab98899accb98295f) as soon as we start working on partial raw query support.

**Reply by antongildebrand** [reply]
Any updates on when this will make it into the roadmap? Being able to have `orderByRaw` would be very valuable for us!
